ccmlib/scylla_node.py

```diff
@@ -1433,6 +1436,57 @@ def do_invoke(sstables):
                 ret[sst] = do_invoke([sst])
         return ret
 
+    def dump_sstables(self,
+                      datafiles: Optional[List[str]] = None,
+                      keyspace: Optional[str] = None,
+                      column_family: Optional[str] = None) -> List[Dict[str, Any]]:
+        """read partitions from specified sstables using `scylla sstable dump-data`
+
+        :param datafiles: paths to sstables (Data components)
+        :param keyspace: restrict the operation to sstables of this keyspace
+        :param column_family: restrict the operation to sstables of this column_family
+        :return: return all the partitions collected in the specified sstables
+        :raises: subprocess.CalledProcessError if scylla-sstable returns a non-zero exit code.
+
+        a typical return value might look like:
+        ```
+        [
+          {
+            'key': {'token': '-4069959284402364209',
+                    'raw': '000400000001',
+                    'value': '1'},
+            'tombstone': {'timestamp': 1690533264324595,
+                          'deletion_time': '2023-07-28 08:34:24z'}
+          },
+          {
+            'key': {'token': '-2249632751995682149',
+                    'raw': '00040000005e',
+                    'value': '94'},
+            'clustering_elements': [
+              {
+                'type': clustering-row',
+                'key': {...},
+                'marker': {...},
+                'columns': {...},
+            ]
+          }
+        ]
+        ```
+        """
+        batch : bool = False
+        column_families: Optional[List[str]] = None
+        if column_family is not None:
+            column_families = [column_family]
+            batch = True
+        sstable_dumps = self.run_scylla_sstable('dump-data', ['--merge'],
+                                                keyspace, datafiles, column_families,
+                                                batch)
+        all_partitions: List[Dict[str, Any]] = []
+        for stdout, _ in sstable_dumps.values():
+            partitions = json.loads(stdout)['sstables']['anonymous']
+            all_partitions += partitions
```

Comment:
Note that this will possibly mash together partitions from different tables. Which might be confusing. I think we should enforce that the user always passes a keyspace and exactly one table parameter. I think this is how most users will use this anyway. The rest (if any) can call `run_scylla_sstable()` instead.

Reply:
changed accordingly.